scene/main/node.cpp

```diff
@@ -1864,8 +1875,12 @@ void Node::_propagate_replace_owner(Node *p_owner, Node *p_by_owner) {
 	data.blocked--;
 }
 
-int Node::get_index() const {
-	return data.pos;
+int Node::get_index(bool p_skip_internal) const {
```

Comment:
This is troublesome because the node may be an internal node and you are asking t to skip itself. It should be checked somewhere and throw an error.